proof/refine/RISCV64/Reply_R.thy

```diff
@@ -894,6 +894,16 @@ lemma updateReply_replyTCB_invs':
           split: option.split_asm)
    by (auto simp: obj_at'_def opt_map_def)
 
+lemma updateReply_replyCanGrant_invs'[wp]:
+  "updateReply rptr (replyCanGrant_update f) \<lbrace>invs'\<rbrace>"
+  apply (simp only: invs'_def valid_pspace'_def)
+  apply (wpsimp wp: updateReply_valid_objs' updateReply_iflive' updateReply_valid_replies' updateReply_list_refs_of_replies')
+  apply (clarsimp simp: valid_replies'_def2 valid_reply'_def live_reply'_def
+                  dest: pspace_alignedD' pspace_distinctD')
+  apply (erule delta_sym_refs)
```

Comment:
I wonder if you could rewrite the `sym_refs` for `list_refs_of_reply'` into the `sym_heap` version we have, and then get this for free.